File: untold_comparison_programs/09_error_handling/go/main.go
package main

import (
    "fmt"
    "os"
    "io/ioutil"
)

func main() {
    fmt.Println("=== Error Handling Demo ===")

    // Success case
    try := func() {
        err := ioutil.WriteFile("test.txt", []byte("Hello"), 0644)
        if err != nil {
            fmt.Println("Error:", err)
            return
        }
        content, err := ioutil.ReadFile("test.txt")
        if err != nil {
            fmt.Println("Error:", err)
            return
        }
        fmt.Printf("Success: %s\n", content)
    }
    try()

    defer func() {
        fmt.Println("Cleanup: deleting file")
        os.Remove("test.txt")
    }()

    // Error case
    defer func() {
        fmt.Println("Done with error demo")
    }()

    _, err := ioutil.ReadFile("nonexistent.txt")
    if err != nil {
        fmt.Printf("Caught error: %v\n", err)
    }
}

File: untold_comparison_programs/09_error_handling/python/main.py
# Error Handling
import os

print("=== Error Handling Demo ===")

# Success case
try:
    with open("test.txt", "w") as f:
        f.write("Hello")
    with open("test.txt", "r") as f:
        content = f.read()
    print(f"Success: {content}")
except Exception as e:
    print(f"Error: {e}")
finally:
    print("Cleanup: deleting file")
    os.remove("test.txt")

# Error case
try:
    with open("nonexistent.txt", "r") as f:
        missing = f.read()
except Exception as e:
    print(f"Caught error: {e}")
finally:
    print("Done with error demo")